sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/JreCertificates.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.Key;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.AccessController;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import static java.util.logging.Level.WARNING;
+
+/**
+ * This class provides the certificates from jre key store. It only provides certificates. It does not provide key entries from jre key store.
+ */
+public final class JreCertificates implements AzureCertificates {
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(JreCertificates.class.getName());
+
+    /**
+     * Stores the jre key store.
+     */
+    private static final KeyStore JRE_KEY_STORE;
+
+    /**
+     * Stores the jre key store aliases.
+     */
+    private static final List<String> ALIASES;
+
+    /**
+     * Stores the jre key store certificates.
+     */
+    private static final  Map<String, Certificate> CERTS;
+
+    /**
+     * Stores the singleton
+     */
+    private static final JreCertificates INSTANCE = new JreCertificates();
+
+    static {
+        JRE_KEY_STORE = JREKeyStore.getDefault();
+        List<String> jreKsAliases = Collections.emptyList();
+        if (null != JRE_KEY_STORE) {
+            try {
+                jreKsAliases = Collections.unmodifiableList(Collections.list(JRE_KEY_STORE.aliases()));
+            } catch (KeyStoreException e) {
+                LOGGER.log(WARNING, "Unable to load the jre key store aliases.", e);
+            }
+        }
+        ALIASES = jreKsAliases;
+        CERTS = ALIASES.stream()
+            .collect(Collectors.toMap(a -> a, a -> {
+                try {
+                    return JRE_KEY_STORE.getCertificate(a);
```

Comment:
And we didn't check null here.

Reply:
> And we didn't check null here.

If JRE_KEY_STORE is null, the ALIASES will be empty list. then  JRE_KEY_STORE.getCertificate(a) won't be executed.